doc/development/twister.md

```diff
@@ -1,59 +1,281 @@
+- `-p PLATFORM, --platform PLATFORM` - Platform filter for testing. This option may be used multiple
+  times. Test suites will only be built/run on the platforms specified. If this option is not used,
+  then platforms marked as default in the platform metadata file will be chosen to build and test.
+  Zephyr tree, so make sure that you always specify it.
+- `-t TAG, --tag TAG` - Specify tags to restrict which tests to run by tag value. Default is to not
+  do any tag filtering. Multiple invocations are treated as a logical 'or' relationship.
+- `-b, --build-only` - Only build the code, do not attempt to run the code on targets.
+- `-f, --only-failed` - Run only those tests that failed the previous twister run invocation.
+- `-y, --dry-run` - Create the filtered list of test cases, but don't actually run them. Useful if
+  you're just interested in the test plan generated for every run and saved in the specified output
+  directory (`testplan.json`).
+- `--overflow-as-errors` - Treat FLASH/RAM/SRAM overflows as errors. **This one is important. If not
+  used the test that overflows is considered to be skipped and not that it has failed.**
+- `-v, --verbose` - Emit build logs, call multiple times to increase verbosity.
+
+## `twister-out` folder
+
+With every invocation of the `east twister` a `twister-out` folder is created. It contains all build
+folders of the created projects, as well as the report files.
+
+Keep in mind, if you keep running `east twister` command and not deleting the `twister-out` folder,
+then the command will rename it to `twister-out.X` (Where `X` is a unique ascending number, starting
+with 1) before running the full build.
+
+## Cookbook style examples
 
-Due to the number of features that it has it can take some time to figure out how to do something
-that you want, see `east twister --extra-help` for the supported options.
+### Running tests on the Host PC
 
-Also, see [Zephyr's documentation](https://docs.zephyrproject.org/latest/develop/test/twister.html)
-for more info.
+You want to build and run all tests under `tests/` directory on the Host PC.
+
+`testcase.yaml` file in the all projects under `tests/` directory should look like this:
+
+```yaml
+tests:
+  tests.test_name.native:
+    platform_allow: native_sim
+    harness: ztest
+```
 
-## Simple example
+**Remember**: Test Scenarios must be unique across the whole repository.
 
 From the project's root directory run:
 
 ```shell
 east twister -p native_sim -T tests
 ```
 
-Twister will detect all projects with `testcase.yaml` in their project root under `tests` directory
-and execute them.
+### Running tests on the target
+
+You want to build and run all tests under `tests/` directory on the target
+
+`testcase.yaml` file in the all projects under `tests/` directory should look like this:
+
+```yaml
+tests:
+  tests.test_name.target:
+    platform_allow: nrf52dk/nrf52832
+    harness: ztest
+```
```

Comment:
Same as above, it sounds like I can either have tests for target only or for native only.
But I can have tests for both, and some tests for only one of them.